### PR TITLE
Adding workflow support for infra deploy

### PR DESCRIPTION
**Pull request by MiguelAngelGutierrezMaya**
### Core changes

- Adds a new `Deploy Infrastructure` workflow to handle CDK deployments for `staging` and `production` from three entry points: successful `Integration Tests` runs, published releases, and manual dispatches.
- Centralizes environment resolution and deploy gating through the shared `resolve_deploy_env` action so automated deployments only proceed when CI passed and relevant paths changed.
- Configures the deploy job to validate required configuration, assume the AWS deployment role through OIDC, support `dry_run` executions with `cdk diff`, and publish CDK outputs as an artifact for downstream workflows.
- Adds a new `Publish API Docs` workflow that runs after a successful infrastructure deployment, exports the OpenAPI definition from API Gateway, bundles Swagger UI assets, and syncs the generated documentation to S3 per environment.

### Schema changes

- No database schema changes.

### Minor changes

- Removes `DSQL_DELETION_PROTECTION` from `config/.env.defaults` because it is no longer part of the active infrastructure configuration.
- Adds per-environment workflow concurrency and job summaries to make deployment runs easier to track and prevent overlapping executions for the same target environment.

## Screenshots/Recordings

Not applicable.

## How to Test

1. Trigger the `Deploy Infrastructure` workflow manually from GitHub Actions with `environment=staging` and `dry_run=true`.
   Expected result: the workflow resolves `staging`, validates the required variables and secrets, assumes the AWS role successfully, and runs `cdk diff` instead of `cdk deploy`.
2. Trigger the same workflow again with `environment=staging` and `dry_run=false` in an environment where the required GitHub environment variables and secrets are already configured.
   Expected result: the deployment completes successfully and uploads the `cdk-outputs-staging` artifact.
3. Verify that `Publish API Docs` starts automatically after the infrastructure workflow finishes successfully.
   Expected result: the workflow downloads the `cdk-outputs-staging` artifact, extracts the API Gateway ID, exports `openapi.json`, and uploads the generated Swagger UI files to the configured S3 prefix.
4. Open the published documentation URL from the workflow summary.
   Expected result: the Swagger UI loads correctly and renders the API definition for the deployed environment.
5. Validate the automatic deploy guard by running the `Integration Tests` workflow on a commit without changes under `infra/`, `services/`, `packages/`, or `.github/workflows/deploy-infra.yml`.
   Expected result: the setup job resolves the environment but marks `should_deploy=false`, so the deploy job is skipped.

## Checklist

- [x] The PR description includes clear instructions on how to test the implementation manually.

##### Added Tests?

- [ ] yes
  - [ ] Verified that this PR is not creating duplicate use cases
  - [ ] All edge cases described in the acceptance criteria are covered, and tests have clear and descriptive names
- [x] no, because they aren't needed
- [ ] no, because I need help

_Override these when needed._

```sh

```

### CODE REVIEW

File: .github/workflows/deploy-infra.yml [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+#     ASSETS_BUCKET_URL, ASSETS_BUCKET_PREFIX, APPLICATION_URL
+#     COGNITO_DOMAIN_PREFIX, COGNITO_CALLBACK_URLS, COGNITO_LOGOUT_URLS
+#     COGNITO_REMOVAL_PROTECT, COGNITO_EMAILS_PREFIX
+#     SES_FROM_EMAIL, SES_REPLY_TO, SNS_MONTHLY_SPEND_LIMIT, SMS_BLOCKED_COUNTRIES
+
+name: Deploy Infrastructure
+
+on:
+  workflow_run:
+    workflows: ['Integration Tests']
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply .github/workflows/deploy-infra.yml:39]: This condition was intentionally removed to test on PRs
